draft-gpew-priv-ppm.md

```diff
@@ -586,33 +600,31 @@ payload = context.Seal(nonce || extensions, input_share)
 where `input_share` is the aggregator's input share and `nonce` and `extensions`
 are the corresponding fields of `Report`.
 
-The leader responds to well-formed requests to `[leader]/upload` with status 200
-and an empty body. Malformed requests are handled as described in {{errors}}.
-Clients SHOULD NOT upload the same measurement value in more than one report if
-the leader responds with status 200 and an empty body.
+An aggregator responds to well-formed requests to `[aggregator]/upload` with
+status 200 and an empty body. Malformed requests are handled as described in
+{{errors}}. Clients SHOULD NOT upload the same measurement value in more than
+one report if the aggregator responds with status 200 and an empty body.
 
-The leader responds to requests with out-of-date `HpkeConfig.id` values,
+The aggregator responds to requests with out-of-date `HpkeConfig.id` values,
 indicated by `EncryptedInputShare.config_id`, with status 400 and an error of
 type 'outdatedConfig'. Clients SHOULD invalidate any cached aggregator
 `HpkeConfig` and retry with a freshly generated Report. If this retried report
 does not succeed, clients MUST abort and discontinue retrying.
 
-The leader MUST ignore any report whose nonce contains a timestamp that falls in
-a batch interval for which it has received at least one collect request from the
-collector. (See {{pa-collect}}.) Otherwise, comparing the aggregate result to
-the previous aggregate result may result in a privacy violation. (Note that the
-helpers enforce this as well; see {{aggregate-request}}.) In addition, the
-leader SHOULD abort the upload protocol and alert the client with error
+To prevent reports from being misused, the aggreagtors filter out replayed or
+otherwise stale reports during the aggregation protocol using the procedure
+described in {{anti-replay}}. In addition, the aggregator that receives a stale
+report SHOULD abort the upload protocol and alert the client with error
 "staleReport".
 
 ### Upload Extensions {#upload-extensions}
```

Comment:
The reason we had upload extensions in the first place was so that they could be tunnelled through the leader into the helper. Now that we do direct uploads, can we remove these? The obvious remaining use case is tunnelling extensions through a client proxy/intermediary/batching client, but if deployments use OHTTP as we recommend, then PPM shouldn't need to specify anything as things like HTTP request headers will get securely relayed.

Reply:
#89 will likely make use of upload extensions, hence we'll need to keep them around.

Reply:
We may also need general extensibility outside of new VDAFs in the future -- a lesson learned from TLS.

Reply:
Well, I don't want to open the can of worms of client attestation in the margins of this PR, but I'm happy to accept TLS as a precedent.